src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -232,49 +247,36 @@ void remove_virtual_functionst::remove_virtual_function(
       t3->make_goto(t_final, true_exprt());
     }
 
-    // Emit target if end of dispatch table is reached or if the next element is
-    // dispatched to another function call. Assumes entries in the functions
-    // variable to be sorted for the identifier of the function to be called.
-    auto l_it = std::next(it);
-    bool next_emit_target =
-      (l_it == functions.crend()) ||
-      l_it->symbol_expr.get_identifier() != fun.symbol_expr.get_identifier();
-
-    // The root function call is done via fall-through, so nothing to emit
-    // explicitly for this.
-    if(next_emit_target && fun.symbol_expr == last_function_symbol)
+    // Fall through to the default callee if possible:
+    if(fallback_action ==
+       virtual_dispatch_fallback_actiont::CALL_LAST_FUNCTION &&
+       fun.symbol_expr == last_function_symbol)
     {
-      class_ids.clear();
+      // Nothing to do
```

Comment:
How about negating the condition to get rid of this branch?

Reply:
I'll keep this as I wanted to be explicit that I'm making a positive "I can skip a branch in exactly this situation" test, rather than a negative test for when I must do work.

Reply:
You could do 
```c++
if(condition)
  continue;

// do the work
```
To make it clear you are skipping over an instruction, removing the unnecessary indentation and make it clear there really is nothing to do in this case